tests/GoogleDriveAdapterTests.php

```diff
@@ -13,11 +15,96 @@ public function stream_open()
 
 class GoogleDriveTests extends PHPUnit_Framework_TestCase
 {
+    // TODO: implement all unit tests
 
-	// TODO: implement all unit tests
+    /**
+     * @var Adapter
+     */
+    protected $adapter;
 
-	public function testTrueIsTrue()
-	{
-		$this->assertTrue(true);
-	}
+    /**
+     * @var PHPUnit_Framework_MockObject_MockObject|Google_Service_Drive
+     */
+    protected $googleDriveService;
+
+    /**
+     * @var PHPUnit_Framework_MockObject_MockObject|Google_Client
+     */
+    protected $defaultGoogleClient;
+
+    /**
+     * @var PHPUnit_Framework_MockObject_MockObject|Google_Http_Batch
+     */
+    protected $defaultGoogleBatch;
+
+    public function setUp()
+    {
+        parent::setUp();
+        $this->defaultGoogleClient = $this->getMockBuilder('Google_Client')
+            ->setMethods(['execute'])
+            ->getMock();
+        $this->defaultGoogleBatch = $this->getMockBuilder('Google_Http_Batch')
+            ->setMethods(['execute'])
+            ->setConstructorArgs([$this->defaultGoogleClient])
+            ->getMock();
+        $this->googleDriveService = $this->getMockBuilder('Google_Service_Drive')
+            ->setMethods(['createBatch'])
+            ->setConstructorArgs([$this->defaultGoogleClient])
+            ->getMock();
+        $this->googleDriveService->files = $this->getMockBuilder('Google_Service_Drive_Resource_Files')->disableOriginalConstructor()->getMock();
+        $this->googleDriveService->expects($this->once())->method('createBatch')->willReturn($this->defaultGoogleBatch);
+        $this->adapter = new Adapter($this->googleDriveService);
+    }
+    /**
+     * @param int $iniBytes
+     * @param int $usedBytes
+     * @param int $fileSizeBytes
+     * @param int $expectedChunkSizeBytes
+     * @dataProvider chunkSizeCalculationDataProvider
+     */
+    public function testChunkSizeCalculation($iniBytes, $usedBytes, $fileSizeBytes, $expectedChunkSizeBytes)
```

Comment:
very nice 🔥

Reply:
well done!  good start to testing this package